rs/ic-observability/multiservice-discovery/tests/tests.rs

```diff
@@ -0,0 +1,144 @@
+// Note: If you need to update the assertions, please check the README.md file for guidelines.
+#[cfg(test)]
+mod tests {
+    use assert_cmd::cargo::CommandCargoExt;
+    use multiservice_discovery_shared::builders::prometheus_config_structure::{
+        PrometheusStaticConfig, IC_NAME, IC_SUBNET, JOB,
+    };
+    use tempfile::tempdir;
+    use std::collections::{BTreeMap, BTreeSet};
+    use std::process::Command;
+    use std::time::Duration;
+    use std::thread;
+    use tokio::runtime::Runtime;
+
+    const BAZEL_SD_BIN: &str = "rs/ic-observability/multiservice-discovery/multiservice-discovery";
+    const CARGO_SD_BIN: &str = "multiservice-discovery";
+
+    async fn fetch_targets() -> anyhow::Result<Vec<PrometheusStaticConfig>> {
+        let timeout_duration = Duration::from_secs(500);
+        let start_time = std::time::Instant::now();
+
+        loop {
+            if start_time.elapsed() > timeout_duration {
+                return Err(anyhow::anyhow!("Timeout reached"));
+            }
+            if let Ok(response) = reqwest::get("http://localhost:8000/prom/targets").await {
+                if response.status().is_success() {
+                    let text = response.text().await?;
+                    let targets: Vec<PrometheusStaticConfig> = serde_json::from_str(&text).unwrap();
+                    return Ok(targets);
+                }
+            }
+            thread::sleep(Duration::from_secs(5));
+        }
+    }
+
+    #[test]
+    fn prom_targets_tests() {
```

Comment:
For asserting things here I suggest that we use the public dashboard API. Here is the [swagger](https://ic-api.internetcomputer.org/api/v3/swagger) of their API and we could fetch all the information from there and check what we need. For eg to get all the nodes just use 
```bash
curl -X 'GET' \
  'https://ic-api.internetcomputer.org/api/v3/nodes?include_status=DOWN&include_status=UP&include_status=UNASSIGNED&include_status=DEGRADED' \
  -H 'accept: application/json'
```
You have an equivalent for subnets as well. Probably easier than to periodically update this test

Reply:
Makes sense @NikolaMilosa ;) thx!